### ISSUE TITLE
Seaweedfs behind NAT broken redirect

### ISSUE THREAD

**Issue by Ne00n** [seaweedfs/seaweedfs#5702]
**Describe the bug**
When running Seaweedfs behind a Reverse Proxy and Seaweedfs itself in a Container, Seaweedfs continues to redirect to its NAT IP. Which breaks any interaction with the API.

**System Setup**
-./weed server -dir=data -ip.bind=0.0.0.0
- Debian 12
- version 30GB 3.67 d3032d1e805ab363242c833d9a61db59b3941f21 linux amd64

**Expected behavior**
It should not redirect any clients to its NAT IP

I tried ip and the mentioned publicURL however none of them are helping to solve these issue.
Does anyone of you got an idea? There needs to be a way to tell Seaweedfs its external IP.

**Comment by chrislusf** [comment]
how is the proxy setup and what is broken with "the API"?

**Comment by Ne00n** [comment]
The Redirect.
Basically I made Seaweedfs reachable over storage.mydomain.com.

Which is hitting a nginx reverse proxy, that passes the traffic down to 10.164.57.174:9333
Thats the container which Seaweedfs is running inside.

However any attempts to upload a file, could not test downloading yet.
Result in a redirect to 10.164.57.174:9333 which is not reachable over the internet.

So any upload attempt will fail.
Any way to tell Seaweedfs not to redirect or redirect properly to the domain mentioned?

**Comment by chrislusf** [comment]
filer is using port 8080. 
9333 is for master.

**Comment by Ne00n** [comment]
So you I should do the upload to the filer with the given fileID from the master?

**Comment by chrislusf** [comment]
No. I do not know what is "the API" you are using to upload.

**Comment by Ne00n** [comment]
I was talking to the master on port 8080 inside the Container.
Did request the fileID on /dir/assign and then started the upload.

Which was also talking to the master, it worked before fine on Localhost.
Without any issues.

However behind a reverse proxy as said above I get redirected to the local IP address inside the container.
As I understand this correctly, I should also expose the filer on port 8080 and use the fileID and start uploading there direct.

However, what if you want to run a cluster, with multiple machines exposed by a reverse proxy?
Would there not be an issue if you run into the same redirect issue again? You don't know what volume server to contact which would usually work fine with the redirect however that doesn't work behind the reverse proxy.

Anyway you can add publicURL back? so the master would provide a correct URL.
Otherwise you probably have to write rewrite rules to make it work behind a reverse proxy properly.

**Comment by chrislusf** [comment]
are you only using the master+volume servers?

**Comment by Ne00n** [comment]
No, as far as I am aware I have currently master + filer + volume running.
I never talked to the volume directly.

Until now I only talked to the master but now also the filer to upload/download files because of the incorrect redirect.
